src/core/dfn.js

```diff
@@ -10,24 +10,90 @@ import {
   showError,
   toMDCode,
 } from "./utils.js";
+import {
+  validateCommonName,
+  validateDOMName,
+  validateMimeType,
+} from "./dfn-validators.js";
 import { registerDefinition } from "./dfn-map.js";
 import { slotRegex } from "./inline-idl-parser.js";
 import { sub } from "./pubsubhub.js";
 
 export const name = "core/dfn";
 
+const knownTypesMap = new Map([
```

Comment:
```suggestion
/** @type {Map<string, { requiresFor: boolean, validator?: (...args: any[])=> boolean,  associateWith?: string}>}  */
const knownTypesMap = new Map([
```
(Might not be entirely correct, please check locally)